### PR TITLE
Spfc 113 resources subcategory

### PR DESCRIPTION
**Pull request by jrzeszutek**
EaaS 2.0 release candidate:
* dev-small, dev-large & production in both AWS & Azure
*  Standalone resources:
  * EKS/AKS
  * RDS/PSQL
  * S3/SA
* v6 environments
* automatically generated unique prefix (eliminates name conflicts)

### CODE REVIEW

File: environments/eaas.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline environments/eaas.yaml:6]: @jrzeszutek shouldn't we assume it will be on master branch ?

File: environments/ttl/enable.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+
+import datetime
+from cloudify import manager
+from cloudify import ctx
+from cloudify.state import ctx_parameters as inputs
+
+SCHEDULE_FORMAT = '%Y-%m-%d %H:%M'
```
Comment (ahmadiesa-abu) [inline environments/ttl/enable.py:8]: @jrzeszutek did we really check the format , as for the documentation the format should be `YYYYMMDDHHMM+HHMM or YYYYMMDDHHMM-HHMM` 
https://docs.cloudify.co/api/v3.1/#start-execution

File: infra/prod/aks/standalone-blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,207 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - plugin:cloudify-azure-plugin?version= >=3.0.10
+  - plugin:cloudify-kubernetes-plugin?version= >=2.9.3
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/aks/standalone-blueprint.yaml:7]: same here we should consider master branch

File: infra/prod/eks/standalone-blueprint.yaml [hunk 4]
```diff
@@ -0,0 +1,259 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - plugin:cloudify-aws-plugin?version= >=2.5.6
+  - plugin:cloudify-kubernetes-plugin?version= >=2.9.3
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/eks/standalone-blueprint.yaml:7]: same here we should consider master branch

File: infra/prod/rds_psql/aws-standalone-blueprint.yaml [hunk 5]
```diff
@@ -0,0 +1,149 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - plugin:cloudify-aws-plugin?version= >=2.5.6
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/rds_psql/aws-standalone-blueprint.yaml:6]: same here we should consider master branch

File: infra/prod/rds_psql/azure-standalone-blueprint.yaml [hunk 6]
```diff
@@ -0,0 +1,128 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - plugin:cloudify-azure-plugin?version= >=3.0.10
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/rds_psql/azure-standalone-blueprint.yaml:6]: same here we should consider master branch

File: infra/prod/s3/aws-blueprint.yaml [hunk 7]
```diff
@@ -1,18 +1,56 @@
 tosca_definitions_version: cloudify_dsl_1_3
 
 imports:
-  - https://cloudify.co/spec/cloudify/5.2.0/types.yaml
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
   - plugin:cloudify-terraform-plugin?version= >=0.15.0
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/s3/aws-blueprint.yaml:6]: same here we should consider master branch

File: infra/prod/s3/azure-blueprint.yaml [hunk 8]
```diff
@@ -1,18 +1,29 @@
 tosca_definitions_version: cloudify_dsl_1_3
 
 imports:
-  - https://cloudify.co/spec/cloudify/5.2.0/types.yaml
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
   - plugin:cloudify-terraform-plugin?version= >=0.15.0
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/s3/azure-blueprint.yaml:6]: same here we should consider master branch

File: infra/prod/s3/azure-standalone-blueprint.yaml [hunk 9]
```diff
@@ -0,0 +1,106 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/6.2.0/types.yaml
+  - plugin:cloudify-terraform-plugin?version= >=0.15.0
+  - https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/custom_types.yaml
```
Comment (ahmadiesa-abu) [inline infra/prod/s3/azure-standalone-blueprint.yaml:6]: same here we should consider master branch

File: infra/prod/s3/azure-blueprint.yaml [hunk 10]
```diff
@@ -24,17 +35,19 @@ node_templates:
...
+          ARM_CLIENT_SECRET: { get_input: [ cloud_credentials, azure_client_secret ] }
         variables:
-          blob_name: { get_input: bucket_name }
-          blob_rg: { concat: [{ get_input: resource_prefix}, 'rg'] }
-          blob_location: { get_input: bucket_region }
+          blob_name: { concat: [ { get_attribute: [ prefix, value ] }, bucket ] }
+          blob_rg: { concat: [ { get_attribute: [ prefix, value ] }, rg ] }
+          blob_location: { get_input: [ cloud_credentials, region_name ] }
         source:
           location: https://github.com/jrzeszutek/tf-blob/archive/master.zip
```
Comment (ahmadiesa-abu) [inline infra/prod/s3/azure-blueprint.yaml:47]: @jrzeszutek do you think we need to make this part of official repos ?

File: infra/rg-tf/blueprint.yaml [hunk 11]
```diff
@@ -45,23 +54,23 @@ node_templates:
...
+          vnet_name: { concat: [ { get_input: [ resource_config, name ] }, '-vnet' ] }
+          vnet_tags: { get_input: [ resource_config, vn_tags ] }
+          cidr_block: { get_input: [ resource_config, cidr_block ] }
+          subnet_1_name: { concat: [ { get_input: [ resource_config, name ] }, '-subnet-1' ] }
+          subnet_1_cidr: { get_input: [ resource_config, subnet_1_cidr ] }
+          subnet_2_name: { concat: [ { get_input: [ resource_config, name ] }, '-subnet-2' ] }
+          subnet_2_cidr: { get_input: [ resource_config, subnet_2_cidr ] }
+          sa_name: { concat: [ { get_input: [ resource_config, name ] }, 'sa' ] }
         source:
           location: https://github.com/jakubcierlik/eaas-terraform/archive/main.zip
```
Comment (ahmadiesa-abu) [inline infra/rg-tf/blueprint.yaml:75]: @jrzeszutek same here , should we consider moving this to official repos ?

File: utils/unique_prefix/custom_types.yaml [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+    properties:
+      predefined_value:
+        type: string
+        description: |
+          If not provided, unique prefix will be generated.
+        required: false
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        start:
+          implementation: https://raw.githubusercontent.com/cloudify-community/eaas-example/SPFC-113-Resources-subcategory/utils/unique_prefix/scripts/generate_prefix.py
```
Comment (ahmadiesa-abu) [inline utils/unique_prefix/custom_types.yaml:18]: same here we should consider master branch

**Review by ahmadiesa-abu — CHANGES_REQUESTED** [review]
left minor comments , more of a concern just to make sure that no accidental removes of branch or repo from non-official repositories could break the blueprints

File: infra/prod/s3/azure-blueprint.yaml (hunk 10 above)
Comment (jrzeszutek) [reply infra/prod/s3/azure-blueprint.yaml:50]: @ahmadiesa-abu I don't have permissions for creating new repos in cloudify-community, so I just moved the zip archives to eaas-example. I think it's not a bad idea to have all of the EaaS code in a single repo

File: infra/rg-tf/blueprint.yaml (hunk 11 above)
Comment (jrzeszutek) [reply infra/rg-tf/blueprint.yaml:75]: @ahmadiesa-abu I don't have permissions for creating new repos in cloudify-community, so I just moved the zip archives to eaas-example. I think it's not a bad idea to have all of the EaaS code in a single repo

File: environments/ttl/enable.py (hunk 2 above)
Comment (jrzeszutek) [reply environments/ttl/enable.py:8]: I don't think it really matters. I tried all validations, I even tried to schedule a workflow manually from the GUI, but it still doesn't appear in the executions list as a scheduled workflow. But workflow is scheduled and it's being triggered at a scheduled time, so maybe let's not change it

**Pull request merged**